src/cc_toolchain/cc_toolchain_config.bzl

```diff
@@ -428,7 +428,7 @@ def _linux_gcc_impl(ctx):
                 flag_groups = [
                     flag_group(flags = ["-fPIC"], expand_if_available = "pic"),
                     flag_group(
-                        flags = ["-fPIE"],
+                        flags = ["-fPIC"],
                         expand_if_not_available = "pic",
                     ),
```

Comment:
is this duplicating the above line? Why was this change needed?

Reply:
Not that it's not the same group: expand_if_not_available vs expand_if_available, but yes I can clean it up

Reply:
Ah, I see the difference, but does this mean that if "pic" is not available, we use the `-fPIC` flag anyways?

Reply:
yes I think so

Reply:
Sorry, I'm still a bit confused here then. Why are we using position independent code when that's not available?

Reply:
"expand_if_not_available: A build variable that needs to be unavailable in order for this flag_group to be expanded."
So in the past we used `fPIE` if the `pic` build variable was not available, but that I don't think that `pic` being unavailable means we can't compile it with `fPIC`. What are your thoughts?

Reply:
How are these build variables set? What does "pic" not being set mean? I'm guessing that either we should fix pic not being set when it should be, or just not use any flag if "pic" is not available, but really all depends on the answer to my first question here.

Reply:
iinm pic is set when it is compiled with our toolchain since it will have `pic_feature` with `name = "pic"`, so the only situations where we have `pic` unavailable is from 3rd party libraries, which should have pic since they should be libraries. I'm not sure if that makes sense and if we can somehow test my hypothesis